dcc-portal-ui/app/scripts/genesets/js/genesets.js

```diff
@@ -247,15 +250,78 @@
                 n.geneId = ensemblId;
               });
             });
+            
+            SetService.getTransientSet('GENE', {
+              'filters': {gene:{pathwayId:{is:[parentPathwayId]}}},
+              'sortBy': 'id',
+              'sortOrder': 'ASCENDING',
+              'name': parentPathwayId,
+              'size': _ctrl.geneSet.geneCount,
+              'transient': true 
+            }).then(function (entitySetData) {
+              CompoundsService.getCompoundsFromEntitySetId(entitySetData.id).then(function(drugs) {                
+                var drugMap = {};
+                _.forEach(drugs, function(drug) {
+                  var zincId = drug.zincId;
+                  var name = drug.name;
+                  var genes = drug.genes;
```

Comment:
(non-blocking) - don't need to extract variables from object properties if they are so short, especially if they're only used once